### DISCUSSION TITLE
Insertion problems in Firefox

### DISCUSSION THREAD

**Discussion by fdw** [fdw/rofimoji#66] [Q&A]
For some reason, Firefox cannot accept Unicode characters with `xdotool`. Thus bug is tracked in https://bugzilla.mozilla.org/show_bug.cgi?id=1657777

(Thanks @katacarbix for this hint 🙂)

**Comment by ar1bis** [comment]
Have you looked into ways of sending unicode characters to an application window from a deeper level than X11 or maybe using ibus? This is something I am trying to figure out, and it would be nice if you could give me any information about alternatives to cut+paste that you might know about.

**Reply by fdw** [reply]
I don't think there is a deeper level than X11, sorry. I haven't looked deeply into ibus, but I'd expect it to rely on X11, as well.
But if you find something, please share 🙂

**Reply by ar1bis** [reply]
IBus seems to do the job. The 'Emoji Picker' on IBus always seems to be able to input unicode chars into whatever window is focused, but xdotool is unable to do this. Let me see if there is some way to send chars from rofimoji into IBus.

**Comment by fdw** [comment]
In the last weeks, it seems to work most of the time. Not reliably all the time, but over 90% of my attempts have worked.
